step4/src/service.js

```diff
@@ -0,0 +1,178 @@
+import {state} from "./state.js";
+import {template} from "./template.js"
+
+function postData(url = '', data = {}) {
+  // Default options are marked with *
+    return fetch(url, {
+        method: 'POST', // *GET, POST, PUT, DELETE, etc.
+        mode: 'cors', // no-cors, cors, *same-origin
+        cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
+        credentials: 'same-origin', // include, *same-origin, omit
+        headers: {
+            'Content-Type': 'application/json',
+            // 'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        redirect: 'follow', // manual, *follow, error
+        referrer: 'no-referrer', // no-referrer, *client
+        body: JSON.stringify(data), // body data type must match "Content-Type" header
+    })
+    .then(response => response.json()); // parses JSON response into native JavaScript objects
+}
+
+/*item 추가 함수 */
+export var addItem = function (event) {
+    event.preventDefault();
+    const $app = document.querySelector('#app');
+    const $appenderForm = $app.querySelector('form[name="appenderForm"]');
+    const content = $appenderForm.querySelector('input').value.trim(); //앞 뒤 빈칸 제거
+    
+    if(content.length === 0) {
+      return alert("enter content");
+    }
+    
+    state.items.push({
+      id: state.items.length,
+      content: $appenderForm.querySelector('input').value,
+      isComplete: false,
+      createtime: Date.now(),
+    })  
+
+    postData('/app/items', {content: content})
+    .then(data => console.log(JSON.stringify(data))) // JSON-string from `response.json()` call
```

Comment:
```suggestion
    .then(console.log) // JSON-string from `response.json()` call
```
간단하게 이렇게 표현해도 된답니다! 물론 JSON.stringify는 적용되지 않겠지만..!

Reply:
준일님 그럼 위의 코드는 ``then(JSON.stringfy).then(console.log)`와도 같은 코드인가요?

Reply:
네 맞습니다.